google/cloud/internal/async_rpc_details.h

```diff
@@ -72,8 +73,14 @@ class AsyncUnaryRpcFuture : public AsyncGrpcOperation {
     ScopedCallContext scope(call_context_);
     if (!ok) {
       // `Finish()` always returns `true` for unary RPCs, so the only time we
-      // get `!ok` is after `Shutdown()` was called; treat that as "cancelled".
-      promise_.set_value(Status(StatusCode::kCancelled, "call cancelled"));
+      // get `!ok` is after `Shutdown()` was called; create a client-origin
+      // "cancelled". By marking the status, we can use that to determine
+      // whether it *might* not be safe to make a GetServerInitialMetadata call
+      // on the client context. This omits the grpc server initial metadata
+      // tracing information from the span attributes.
```

Comment:
I'd probably leave the GetServerInitialMetadata()/span-attributes part of this comment to the code that checks for the "origin=client".

Reply:
Done

Reply:
done